### DISCUSSION TITLE
Point render similar to other types

### DISCUSSION THREAD

**Discussion by malstraem** [Mapsui/Mapsui#1462] [Q&A] [answered]
Hi, how can I use point geometry as well as polygons with scaling? Should I create my own IStyle and ISkiaStyleRenderer classes?

**Comment by pauldendulk** [comment]
Not sure what you mean. Do you want the size of the point symbol to grow when you zoom in?

**Reply by malstraem** [reply]
Yes, I want the points to be scalable and appear when they are actually visible, not a symbol behavior. I used a vector style for layer to display the shapefile on the map. But points behave like symbols - always visible and static size.

And this code form `PointRenderer` some confused me
```
else if (style is VectorStyle vectorStyle)
{
      // Use the SymbolStyleRenderer and specify Ellipse
      SymbolStyleRenderer.Draw(canvas, vectorStyle, destination, opacity, SymbolType.Ellipse);
}
```

**Comment by pauldendulk** [comment, marked as answer]
Okay. I needed that scenario myself a long time ago. Currently this is not supported off the shelf.

Note, that you probably will also need a regular (non scaling) style for when you are zoomed out. You can use multiple styles on one feature. And you can show them depending on zoom level (Resolution).

Options:
- You could create a custom style renderer: https://mapsui.com/documentation/custom-style-renders.html
- Don' use a symbols but actually create a geometry (this does not allow use of bitmaps).
- Use a Raster feature (like used in the tiles and the image layers). You will need to derive the extent for each feature from the point location.

**Reply by malstraem** [reply]
Much thanks
